### PR TITLE
SASS-1887b  Adding buttons to complete in year employment journey

### PR DESCRIPTION
**Pull request by dnyuar**
### Description
Adding buttons to check your employment details, expenses and benefits pages for in year

Journey tests (Need to be merged first) -> hmrc/income-tax-submission-journey-tests#115

Add a link to the relevant story in Jira
[SASS-1357](https://jira.tools.tax.service.gov.uk/browse/SASS-1357)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/views/employment/CheckEmploymentDetailsView.scala.html [hunk 1]
```diff
@@ -38,6 +39,28 @@
...
+            href = Some(controllers.employment.routes.EmploymentSummaryController.show(taxYear).url)
+        ))
+            } else {
+                govUkButton(Button(
+                    attributes = Map("id" -> "returnToEmployerBtn"),
+                    preventDoubleClick = true,
+                    content = Text(messages("common.returnToEmployer")),
+                    classes = "govuk-button",
+                    href = Some(controllers.employment.routes.EmploymentDetailsAndBenefitsController.show(taxYear, employmentDetails.employmentId).url)
+                ))
```
Comment (BenCosfordHMRC) [inline app/views/employment/CheckEmploymentDetailsView.scala.html:59]: could use our button helper instead here

File: app/views/employment/CheckEmploymentDetailsView.scala.html [hunk 2]
```diff
@@ -118,6 +141,8 @@
         @formWithCSRF(action = controllers.employment.routes.CheckEmploymentDetailsController.submit(taxYear, employmentDetails.employmentId)) {
             @button("common.saveAndContinue", classes = Some("govuk-!-margin-top-6"))
         }
+    } else {
+        @employmentSource
```
Comment (BenCosfordHMRC) [inline app/views/employment/CheckEmploymentDetailsView.scala.html:145]: and can use the formHelper with the button helper here
like https://github.com/hmrc/income-tax-employment-frontend/blob/main/app/views/templates/TimeoutPage.scala.html#L33

File: app/views/employment/CheckYourBenefitsView.scala.html [hunk 3]
```diff
@@ -215,6 +238,10 @@ <h2 class="govuk-label--m govuk-!-margin-top-9">@messages("checkYourBenefits.ass
             govukSummaryList(SummaryList(rows))
         }
     }
+
+    @if(isInYear) {
+        @employmentSource
```
Comment (BenCosfordHMRC) [inline app/views/employment/CheckYourBenefitsView.scala.html:243]: same as above

File: app/views/expenses/CheckEmploymentExpensesView.scala.html [hunk 4]
```diff
@@ -79,6 +80,17 @@
...
+    @if(isInYear) {
+
+        @govUkButton(Button(
+            attributes = Map("id" -> "returnToEmploymentSummaryBtn"),
+            preventDoubleClick = true,
+            content = Text(messages("employment.employmentInfo.buttonText")),
+            classes = "govuk-button",
+            href = Some(controllers.employment.routes.EmploymentSummaryController.show(taxYear).url)
+        ))
+    }
```
Comment (BenCosfordHMRC) [inline app/views/expenses/CheckEmploymentExpensesView.scala.html:92]: as above

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

File: app/controllers/employment/CheckEmploymentDetailsController.scala [hunk 5]
```diff
@@ -91,8 +91,11 @@ class CheckEmploymentDetailsController @Inject()(implicit val cc: MessagesContro
             if (!cya.isPriorSubmission && !cya.employment.employmentDetails.isFinished) {
               Future.successful(RedirectService.employmentDetailsRedirect(cya.employment, taxYear, employmentId, cya.isPriorSubmission))
             } else {
-              Future.successful(performAuditAndRenderView(cya.employment.toEmploymentDetailsView(
-                employmentId, !cya.employment.employmentDetails.currentDataIsHmrcHeld), taxYear, isInYear))
+              prior match {
+                case Some(employment) => Future.successful(performAuditAndRenderView(cya.employment.toEmploymentDetailsView(
+                  employmentId, !cya.employment.employmentDetails.currentDataIsHmrcHeld), taxYear, isInYear, employment))
+                case None => Future.successful(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
```
Comment (robert-webb-hmrc) [inline app/controllers/employment/CheckEmploymentDetailsController.scala:97]: I think we need to check the outcome of this function, as we are now seeing failures in the journey tests relating to the match of prior data here.

Comment (BenCosfordHMRC) [reply app/controllers/employment/CheckEmploymentDetailsController.scala:97]: yeah prior data is not mandatory here so shouldn't be redirecting if no prior data

File: app/controllers/employment/CheckYourBenefitsController.scala [hunk 6]
```diff
@@ -111,11 +111,15 @@ class CheckYourBenefitsController @Inject()(authorisedAction: AuthorisedAction,
...
+                                employmentId: String, isUsingCustomerData: Boolean, cya: Option[BenefitsViewModel] = None, allEmploymentData: AllEmploymentData)(implicit user: User[AnyContent]): Result = {
     val auditModel = ViewEmploymentBenefitsAudit(taxYear, user.affinityGroup.toLowerCase, user.nino, user.mtditid, benefits)
     auditService.sendAudit[ViewEmploymentBenefitsAudit](auditModel.toAuditModel)
+
+    val employmentSource: Seq[EmploymentSource] = employmentSessionService.getLatestEmploymentData(allEmploymentData, isInYear)
+    val isSingleEmployment: Boolean = employmentSource.length == 1
+
     if (isInYear) {
-      Ok(checkYourBenefitsView(taxYear, benefits))
+      Ok(checkYourBenefitsView(taxYear, benefits, isSingleEmployment, isInYear, employmentId))
```
Comment (BenCosfordHMRC) [inline app/controllers/employment/CheckYourBenefitsController.scala:122]: dont need to pass isInYear here as that view is only for in year

**Comment by robert-webb-hmrc** [comment]
Currently failing tests

**Review by BenCosfordHMRC — APPROVED** [review]

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**